src/prefect/tasks.py

```diff
@@ -1059,6 +1031,116 @@ def map(
+    ) -> TaskRun:
+        ...
+
+    @overload
+    def apply_async(
+        self: "Task[P, Coroutine[Any, Any, T]]",
+        *args: P.args,
+        **kwargs: P.kwargs,
+    ) -> Awaitable[TaskRun]:
+        ...
+
+    def apply_async(
+        self,
+        *args: Any,
+        **kwargs: Any,
+    ) -> Union[TaskRun, Awaitable[TaskRun]]:
+        """
+        Create a pending task run for a task server to execute.
+
+        If writing an async task, this call must be awaited.
+
+        Args:
+            *args: Arguments to run the task with
+            **kwargs: Keyword arguments to run the task with
+
+        Returns:
+            A TaskRun object representing the pending task run
+
+        Examples:
+
+            Define a task
+
+            >>> from prefect import task
+            >>> @task
+            >>> def my_task():
+            >>>     return "hello"
+
+            Create a pending task run for the task
+
+            >>> from prefect import flow
+            >>> @flow
+            >>> def my_flow():
+            >>>     my_task.apply_async()
+
+            TODO: Wait for a task to finish
+
+            >>> @flow
+            >>> def my_flow():
+            >>>     my_task.apply_async().wait()  # <- This is not implemented
+
+
+            >>> @flow
+            >>> def my_flow():
+            >>>     print(my_task.apply_async().result())  # <- This is not implemented
+            >>>
+            >>> my_flow()
+            hello
+
+            Create a pendingn task in an async flow
+
+            >>> @task
+            >>> async def my_async_task():
+            >>>     pass
+            >>>
+            >>> @flow
+            >>> async def my_flow():
+            >>>     await my_async_task.apply_async()
+
+            TODO: Enforce ordering between tasks that do not exchange data
+            >>> @task
+            >>> def task_1():
+            >>>     pass
+            >>>
+            >>> @task
+            >>> def task_2():
+            >>>     pass
+            >>>
+            >>> @flow
+            >>> def my_flow():
+            >>>     x = task_1.apply_async()
+            >>>
+            >>>     # task 2 will wait for task_1 to complete
+            >>>     y = task_2.apply_async(wait_for=[x])  # <- Not implemented, unsure if will
+
+        """
+
+        # TODO: Consolidate these into a single function that can create a task run
+        # and store parameters with or without a flow run ID.
+        from prefect.engine import (
+            create_autonomous_task_run,
+            create_task_run,
+        )
+
+        # Convert the call args/kwargs to a parameter dict
+        parameters = get_call_parameters(self.fn, args, kwargs)
+        flow_run_context = FlowRunContext.get()
+
+        # Create a pending task run
+        create_fn = create_task_run if flow_run_context else create_autonomous_task_run
+        create_task_run_call = create_call(create_fn, task=self, parameters=parameters)
```

Comment:
is using the current engine primitives and utilities necessary for some reason?

Reply:
I'm going to see how much I can remove before merging.